File: Searching/Binary Search/Python/BinarySearch.py
"""
This is a simple binary search algorithm for python.
This is the exact implementation of the pseudocode written in README.
"""

def binary_search(item_list,item):
    left_index = 0
    right_index = len(item_list)-1
    
    while left_index <= right_index:
        middle_index = (left_index+right_index) / 2
        if item_list[middle_index] < item:
            left_index=middle_index+1
        elif item_list[middle_index] > item:
            right_index=middle_index-1
        else:
            return middle_index
    return None

#Simple demonstration of the function
if __name__=='__main__':
    print (binary_search([1,3,5,7],3))
    print (binary_search([1,2,5,7,97],0))

File: Searching/Binary Search/PHP/BinarySearch.php
<?php

    function binarySearch($list, $item){
        $leftIndex = 0;
        $rightIndex = sizeof($list);
        
        while($leftIndex <= $rightIndex){
            $middleIndex = intdiv(($leftIndex + $rightIndex), 2);
            
            if($list[$middleIndex] < $item){
                $leftIndex = $middleIndex + 1;
            }
            else if ($list[$middleIndex] > $item){
                $rightIndex = $middleIndex - 1;
            }
            else{
                return $middleIndex;
            }
        }
        return -1;
    }
    
    // Simple demonstration of the function
    $array = array(1, 2, 4, 5, 6, 8, 9, 10, 14);
    echo binarySearch($array, 8);
?>
